File: src/Library.java
import java.util.ArrayList;
import java.util.List;

public class Library {
    private List<Book> books;
    private List<User> users;

    public Library(){
        this.books=new ArrayList<>();
        this.users=new ArrayList<>();
    }

    public void addBook(String title, String author) {
        books.add(new Book(title, author));
        System.out.println("Book added with success.");
    }

    public void addUser(String name) {
        users.add(new User(name));
        System.out.println("User added with success.");
    }

    public void rentBook(String title, String author, String user) {
        boolean rented=false;
        boolean findUser=false;
        for(User user1 : users) {
            if(user.equals(user1.getName())) {
                findUser=true;
                for (Book book : books) {
                    if(title.equals(book.getTitle()) && !book.isRented()) {
                        book.setRented(true);
                        user1.rentBook(book.getId(), title, author);
                        rented=true;
                        System.out.println("Book rented.");
                    }
                }
                if(!rented)
                    System.out.println("Book wasn't found, or it has been rented.");
            }
        }
        if(!findUser)
            System.out.println("User wasn't found.");
    }

    public void printUsers() {
        String userInfo;
        for (User user : users) {
            System.out.println("------------------------------");
            userInfo = "";
            userInfo += "User ID: " + (user.getId()) + "\n";
            userInfo += "Name: " + user.getName() + "\n";
            System.out.println(userInfo);
            userInfo = "";
            userInfo += user.getUserBooks(user.getName());
            System.out.println(userInfo);
        }
        System.out.println("------------------------------");
    }

    public void printUserWithBooks() {
        String userInfo;
        for (User user : users) {
            if(!user.getBooks().isEmpty()) {
                userInfo = "";
                System.out.println("------------------------------");
                System.out.println("User ID: " + (user.getId()));
                userInfo += "Name: " + user.getName();
                userInfo += user.getUserBooks(user.getName());
                System.out.println(userInfo);
            }
        }
        System.out.println("------------------------------");
    }

    public void printBooks() {
        String bookInfo;
        for (Book book : books) {
            System.out.println("------------------------------");
            bookInfo = "";
            System.out.println("Book's ID: " + (book.getId()) + ":");
            bookInfo += "Book's title: " + book.getTitle() + "\n";
            bookInfo += "Book's author: " + book.getAuthor();
            System.out.println(bookInfo);
        }
        System.out.println("------------------------------");
    }

    public void printRentedBooks() {
        String bookInfo;
        if(books.isEmpty()) {
            System.out.println("There's no rented books.");
        } else {
            for (Book book : books) {
                if(book.isRented()) {
                    System.out.println("------------------------------");
                    bookInfo = "";
                    System.out.println("Book's ID: " + (book.getId()) + ":");
                    bookInfo += "Book's title: " + book.getTitle() + "\n";
                    bookInfo += "Book's author: " + book.getAuthor();
                    System.out.println(bookInfo);
                }
            }
            System.out.println("------------------------------");
        }
    }

    public void printUnrentedBooks() {
        String bookInfo;
        for (Book book : books) {
            if(!book.isRented()) {
                System.out.println("------------------------------");
                bookInfo = "";
                System.out.println("Book's ID: " + (book.getId()) + ":");
                bookInfo += "Book's title: " + book.getTitle() + "\n";
                bookInfo += "Book's author: " + book.getAuthor();
                System.out.println(bookInfo);
            }
        }
        System.out.println("------------------------------");
    }

    public List<User> getUsers() {
        return users;
    }

    public List<Book> getBooks() {
        return books;
    }

}

File: library.py
from book import Book
from user import User

class Library:
    def __init__(self):
        self.books = []
        self.users = []

    def addBook(self, title, author):
        self.books.append(Book(title, author))
        print("Book added with success.")

    def addUser(self, name):
        self.users.append(User(name))
        print("User added with success.")

    def rentBook(self, title, author, user):
        rented = False
        findUser = False
        for user_obj in self.users:
            if user == user_obj.getName():
                findUser = True
                for book in self.books:
                    if title == book.getTitle() and not book.isRented():
                        book.setRented(True)
                        user_obj.rentBook(book.getId(), title, author)
                        rented = True
                        print("Book rented.")
                if not rented:
                    print("Book wasn't found, or it has been rented.")
        if not findUser:
            print("User wasn't found.")

    def printUsers(self):
        for user in self.users:
            print("------------------------------")
            print(f"User ID: {user.getId()}")
            print(f"Name: {user.getName()}")
            user_books_info = user.getUserBooks(user.getName())
            if user_books_info:
                print(user_books_info)
        print("------------------------------")

    def printUserWithBooks(self):
        for user in self.users:
            if user.getBooks():
                print("------------------------------")
                print(f"User ID: {user.getId()}")
                print(f"Name: {user.getName()}")
                user_books_info = user.getUserBooks(user.getName())
                if user_books_info:
                    print(user_books_info)
        print("------------------------------")

    def printBooks(self):
        for book in self.books:
            print("------------------------------")
            print(f"Book's ID: {book.getId()}:")
            print(f"Book's title: {book.getTitle()}")
            print(f"Book's author: {book.getAuthor()}")
        print("------------------------------")

    def printRentedBooks(self):
        rented_books = [book for book in self.books if book.isRented()]
        if not rented_books:
            print("There's no rented books.")
        else:
            for book in rented_books:
                print("------------------------------")
                print(f"Book's ID: {book.getId()}:")
                print(f"Book's title: {book.getTitle()}")
                print(f"Book's author: {book.getAuthor()}")
            print("------------------------------")

    def printUnrentedBooks(self):
        unrented_books = [book for book in self.books if not book.isRented()]
        for book in unrented_books:
            print("------------------------------")
            print(f"Book's ID: {book.getId()}:")
            print(f"Book's title: {book.getTitle()}")
            print(f"Book's author: {book.getAuthor()}")
        print("------------------------------")

    def getUsers(self):
        return self.users

    def getBooks(self):
        return self.books
